src/fmu/dataio/_metadata.py

```diff
@@ -156,7 +159,9 @@ def _populate_meta_objectdata(self) -> None:
 
         Hence this must be ran early or first.
         """
-        self.objdata = ObjectDataProvider(self.obj, self.dataio, self.meta_existing)
+        self.objdata = objectdata_provider_factory(
+            self.obj, self.dataio, self.meta_existing
+        )
```

Comment:
We could here have used the opportunity to make more explicit  input to the function, i.e. avoid the `self.dataio` object but rather distillate exactly what to be transferred, similar to what I did for `FmuProvider` a while ago. Or we can take in the next PR...

Reply:
Good point, I think that's a good idea. I expect to be focusing on the providers for a bit; I will keep this in mind. Maybe there is something we can do about this between all of the providers.